winpr/libwinpr/ncrypt/ncrypt_pkcs11.c

```diff
@@ -583,15 +614,29 @@ static SECURITY_STATUS NCryptP11KeyGetProperties(NCryptP11KeyHandle* keyHandle,
 	CK_ATTRIBUTE* objectFilter = certificateFilter;
 	CK_ULONG objectFilterLen = ARRAY_LENGTH(certificateFilter);
 
-	if (property == NCRYPT_PROPERTY_UNKNOWN)
-		return NTE_NOT_SUPPORTED;
-
-	/* TODO: shall adjust objectFilter and objectFilterLen depending on requested
-	 * the property when we add new ones */
 	switch (property)
 	{
 		case NCRYPT_PROPERTY_CERTIFICATE:
 			break;
+		case NCRYPT_PROPERTY_READER: {
+			CK_SLOT_INFO slotInfo;
+
+			rv = provider->p11->C_GetSlotInfo(keyHandle->slotId, &slotInfo);
```

Comment:
Better assert pointer dereferences here to make tracing issues easier

Reply:
I don't see what is more likely to happen here that would require an assert. And BTW segfaulting here would make a clean and nice segfault easily debuggable.